dialect_mysql.go

```diff
@@ -80,10 +81,18 @@ func (mysql) DataTypeOf(field *StructField) string {
 			}
 		case reflect.Struct:
 			if _, ok := dataValue.Interface().(time.Time); ok {
-				if _, ok := field.TagSettings["NOT NULL"]; ok {
```

Comment:
The "NOT NULL" tag checking is removed because I don't see it is necessary here.  Caller will properly insert NOT NULL if tag exists.  The new test case added below also confirms "not null" tag works as expected.